subsys/net/ip/tcp2.c

```diff
@@ -989,8 +992,6 @@ int net_tcp_put(struct net_context *context)
 		conn_state(conn, TCP_FIN_WAIT_1);
 	}
 
-	net_context_unref(context);
```

Comment:
You are missing this in your ozhuraki@a25b55d.

Reply:
@rveerama1 

> You are missing this in your ozhuraki/zephyr@a25b55d

The net_context needs to release its' reference. The problem might be somewhere else then.

Reply:
ok